src/main/java/de/tum/in/www1/artemis/repository/ParticipationRepository.java

```diff
@@ -1,10 +1,25 @@
 package de.tum.in.www1.artemis.repository;
 
+import java.util.List;
+
+import org.springframework.data.jpa.repository.EntityGraph;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
 import org.springframework.stereotype.Repository;
 
 import de.tum.in.www1.artemis.domain.Participation;
 
 @Repository
 public interface ParticipationRepository extends JpaRepository<Participation, Long> {
+
+    @Query("select distinct pe from Participation pe left join fetch pe.submissions where pe.id = :#{#participationId}")
+    Participation getOneWithEagerSubmissions(@Param("participationId") Long participationId);
+
+    @Query("select pe from Participation pe left join fetch pe.submissions")
+    List<Participation> getAllWithEagerSubmissions();
+
+    @EntityGraph(attributePaths = { "submissions", "results.submission" })
+    @Query("select pe from Participation pe")
```

Comment:
I guess you can leave out the `@Query` annotation here, because you have the `@EntityGraph` annotation

Reply:
The problem is here that if I remove the `@Query` that the method name can't be parsed:
`Failed to create query for method public abstract java.util.List de.tum.in.www1.artemis.repository.ParticipationRepository.getAllWithEagerSubmissionsAndResults()! No property getAllWithEagerSubmissions found for type Participation!`

It seemed to me as the nicest compromise for this use case as we don't have to list the `left join fetch` in the query.